js/privatebin.js

```diff
@@ -4972,7 +5033,12 @@ jQuery.PrivateBin = (function($, RawDeflate) {
 
             // prepare server interaction
             ServerInteraction.prepare();
-            ServerInteraction.setCryptParameters(TopNav.getPassword());
+            if(!TopNav.getBurnAfterReading()) {
+                ServerInteraction.setCryptParameters(TopNav.getPassword());
+            } else {
+                // not needed in this scenario
+                ServerInteraction.setCryptParameters('');
```

Comment:
A thing to discuss is whether we maybe would still also have the server-side password authentication in place hmm?

~~(BTW totally forgot how that is made, is the password really saved on the server? If so, this approach is would of course be more zero-knowledge, because we would not have to save such sensitive data server-side.)~~

Reply:
Hm, it would be nice if we could pass the now confirmed password to the new loaded paste URL. But that risks leaking it in yet another way.

On the plus side, yes, the recipient only ever gets to see the true URL and key upon entering the password correctly. Unlike for any other paste type, where the key is leaked in the shared URL to anyone reading. Which is why we recommend a password in the first place. Here, the visible key plus password unlock the hidden key.